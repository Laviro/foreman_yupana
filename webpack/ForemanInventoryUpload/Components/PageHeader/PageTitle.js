import React, { useState } from 'react';
import {
  Dropdown,
  DropdownItem,
  KebabToggle,
  DropdownPosition,
} from '@patternfly/react-core';
import {
  INVENTORY_PAGE_TITLE,
  ACTIONS_HISTORY_BUTTON_TEXT,
  DOCS_BUTTON_TEXT,
} from '../../ForemanInventoryConstants';
import {
  getActionsHistoryUrl,
  getInventoryDocsUrl,
} from '../../ForemanInventoryHelpers';

const PageTitle = () => {
  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
  const dropdownItems = [
    <DropdownItem
      key="tasks-history-button"
      href={getActionsHistoryUrl()}
      target="_blank"
      rel="noopener noreferrer"
    >
      {ACTIONS_HISTORY_BUTTON_TEXT}
    </DropdownItem>,
    <DropdownItem
      key="inventory-documentation-button"
      href={getInventoryDocsUrl()}
      target="_blank"
      rel="noopener noreferrer"
    >
      {DOCS_BUTTON_TEXT}
    </DropdownItem>,
  ];
  return (
    <div className="row form-group inventory-upload-header-title">
      <h1 className="col-md-8">{INVENTORY_PAGE_TITLE}</h1>
      <Dropdown
        className="title-dropdown"
        onSelect={() => setIsDropdownOpen(false)}
        toggle={<KebabToggle onToggle={isOpen => setIsDropdownOpen(isOpen)} />}
        isOpen={isDropdownOpen}
        isPlain
        dropdownItems={dropdownItems}
        position={DropdownPosition.right}
      />
    </div>
  );
};
export default PageTitle;
