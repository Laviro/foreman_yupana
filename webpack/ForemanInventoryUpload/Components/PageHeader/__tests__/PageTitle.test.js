import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import PageTitle from '../PageTitle';

const fixtures = {
  'render without Props': {},
};

describe('PageTitle', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(PageTitle, fixtures));
});
