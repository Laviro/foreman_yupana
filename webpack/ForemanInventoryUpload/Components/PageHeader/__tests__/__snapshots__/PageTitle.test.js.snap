// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle rendering render without Props 1`] = `
<div
  className="row form-group inventory-upload-header-title"
>
  <h1
    className="col-md-8"
  >
    Red Hat Inventory
  </h1>
  <Dropdown
    className="title-dropdown"
    dropdownItems={
      Array [
        <DropdownItem
          href="/foreman_tasks/tasks?search=action++%3D++ForemanInventoryUpload%3A%3AAsync%3A%3AGenerateReportJob+or+action++%3D++ForemanInventoryUpload%3A%3AAsync%3A%3AGenerateAllReportsJob&page=1"
          rel="noopener noreferrer"
          target="_blank"
        >
          Actions history
        </DropdownItem>,
        <DropdownItem
          href="/links/manual/+?root_url=https%3A%2F%2Faccess.redhat.com%2Fproducts%2Fsubscription-central"
          rel="noopener noreferrer"
          target="_blank"
        >
          Documentation
        </DropdownItem>,
        <DropdownItem
          onClick={[Function]}
        >
          Connectivity test
        </DropdownItem>,
      ]
    }
    isOpen={false}
    isPlain={true}
    onSelect={[Function]}
    position="right"
    toggle={
      <KebabToggle
        onToggle={[Function]}
      />
    }
  />
  <CloudPingModal
    isOpen={false}
    title="Connectivity test"
    toggle={[Function]}
  />
</div>
`;
